core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournal.java

```diff
@@ -217,11 +225,24 @@ public synchronized void start() throws IOException {
    */
   public synchronized void gainPrimacy() throws IOException {
     Preconditions.checkState(mWriter == null, "writer must be null in secondary mode");
-    Preconditions.checkState(mTailerThread != null,
+    Preconditions.checkState(mSuspended || mTailerThread != null,
         "tailer thread must not be null in secondary mode");
-    mTailerThread.awaitTermination(true);
-    long nextSequenceNumber = mTailerThread.getNextSequenceNumber();
-    mTailerThread = null;
+
+    long nextSequenceNumber;
+    if (mSuspended) {
+      // Cancel if advancing is in progress.
+      if (mAdvanceFuture != null && !mAdvanceFuture.isDone()) {
+        // TODO(ggezer): Avoid re-applying sequences read while advancing.
```

Comment:
Did we stop backing up?

Reply:
Yes, backup will be cancelled if backup-worker fail over to become a leader.